sdk/core/azure-core/src/main/java/com/azure/core/implementation/ResponseConstructorsCache.java

```diff
@@ -117,7 +120,8 @@
                                 bodyAsObject,
                                 decodedHeaders);
                         } catch (IllegalAccessException | InvocationTargetException | InstantiationException e) {
-                            throw logger.logExceptionAsError(Exceptions.propagate(e));
+                            throw logger.logExceptionAsError(new RuntimeException("Failed to deserialize 5-parameter"
+                                + "response with decoded headers. Error Details: " + e.getMessage(), e));
```

Comment:
```suggestion
                                + " response with decoded headers. Error Details: " + e.getMessage(), e));
```

Reply:
The downside of doing string concatenations like this where it is split between multiple lines is the actual string may not be formatted (ie. missing spaces) correctly.

Reply:
This suggestion also applies to the other messages above.